### PR TITLE
Add support for byte[] properties

### PR DESCRIPTION
**Pull request by fealebenpae**
Implements #510

### CODE REVIEW

File: Tests/IntegrationTests.Shared/AccessTests.cs [hunk 1]
```diff
@@ -71,5 +95,19 @@ public void SetValueAndReplaceWithNull(string propertyName, object propertyValue
 
             Assert.That(TestHelpers.GetPropertyValue(ato, propertyName), Is.EqualTo(null));
         }
+
+        public IEnumerable<object[]> SetAndReplaceWithNullCases()
+        {
+            yield return new object[] { "NullableCharProperty", '0' };
```
Comment (AndyDentFree) [inline Tests/IntegrationTests.Shared/AccessTests.cs]: Looks like copy and paste error - shouldn't these all be null?

File: Tests/IntegrationTests.Shared/AccessTests.cs [hunk 2]
```diff
@@ -41,15 +44,36 @@ public void TearDown()
...
+        {
+            yield return new object[] { "CharProperty", '0' };
+            yield return new object[] { "ByteProperty", (byte)100 };
+            yield return new object[] { "Int16Property", (short)100 };
+            yield return new object[] { "Int32Property", 100 };
+            yield return new object[] { "Int64Property", 100L };
+            yield return new object[] { "SingleProperty", 123.123f };
+            yield return new object[] { "DoubleProperty", 123.123 };
+            yield return new object[] { "BooleanProperty", true };
+            yield return new object[] { "ByteArrayProperty", new byte[] { 0xde, 0xad, 0xbe, 0xef } };
```
Comment (AndyDentFree) [inline Tests/IntegrationTests.Shared/AccessTests.cs]: Also would like a case to clarify that an array of length zero is not null.

File: Tests/IntegrationTests.Shared/AccessTests.cs (hunk 1 above)
Comment (fealebenpae) [inline Tests/IntegrationTests.Shared/AccessTests.cs]: This test first sets the property to a non-null value and asserts the getter returns the value, and then sets the property to null and asserts that the getter returns null.

Comment (AndyDentFree) [inline Tests/IntegrationTests.Shared/AccessTests.cs]: OK, yup noted it now.

**Comment by AndyDentFree** [comment]
👍

File: Tests/IntegrationTests.Shared/AccessTests.cs (hunk 2 above)
Comment (fealebenpae) [inline Tests/IntegrationTests.Shared/AccessTests.cs]: Good catch, that needed additional work - see latest commit.

File: Realm.Shared/linq/RealmResultsVisitor.cs [hunk 3]
```diff
@@ -363,6 +363,24 @@ private static void AddQueryEqual(QueryHandle queryHandle, string columnName, ob
                 NativeQuery.double_equal((QueryHandle)queryHandle, columnIndex, (double)value);
             else if (valueType == typeof(DateTimeOffset))
                 NativeQuery.datetime_seconds_equal(queryHandle, columnIndex, ((DateTimeOffset)value).ToUnixTimeSeconds());
+            else if (valueType == typeof(byte[]))
+            {
+                var buffer = (byte[])value;
+                if (buffer.Length == 0)
+                {
+                    // see RealmObject.SetByteArrayValue
+                    NativeQuery.binary_equal(queryHandle, columnIndex, (IntPtr)0x1, IntPtr.Zero);
```
Comment (AndyDentFree) [inline Realm.Shared/linq/RealmResultsVisitor.cs]: I'm bothered by this `0x1` magic value being repeated as a literal. Does it have significance and, if so, could it be a constant?

Comment (fealebenpae) [inline Realm.Shared/linq/RealmResultsVisitor.cs]: No, it just has to be anything but zero, as long as the length is zero. Seeing as how it's a dummy value I wouldn't bother with a constant.

**Comment by AndyDentFree** [comment]
👍 with the added check for empty byte arrays!

**Comment by kristiandupont** [comment]
:+1:

**Pull request merged**